File: sdk/typescript-guest/templates/word-count.ts
import type * as Contract from '../generated/interfaces/examples-word-count-api.js';
import { whitespace, utf8Length } from '../vendor/lsf/sdk/typescript-guest/runtime/text.js';

// lsf-example-begin: capsule
export const api: typeof Contract = {
  count(text) {
    if (utf8Length(text) > 4096) throw 'Use text of at most 4096 bytes.';
    let count = 0, inWord = false;
    for (const scalar of text) {
      const word = !whitespace(scalar);
      if (word && !inWord) count++;
      inWord = word;
    }
    return count;
  },
};
// lsf-example-end: capsule

File: sdk/dotnet-guest/templates/word-count.cs
using ServiceWorld;
using Lsf.Guest;
namespace ServiceWorld.wit.Exports.examples.wordCount;

// lsf-example-begin: capsule
public class ApiExportsImpl : IApiExports
{
    public static Result<uint, string> Count(string text)
    {
        if (Text.Utf8Length(text) > 4096) return Result<uint, string>.Err("Use text of at most 4096 bytes.");
        uint count = 0;
        bool inWord = false;
        foreach (char scalar in text) {
            bool word = !Text.Whitespace(scalar);
            if (word && !inWord) ++count;
            inWord = word;
        }
        return Result<uint, string>.Ok(count);
    }
}
// lsf-example-end: capsule
